crates/wast/src/lib.rs

```diff
@@ -172,6 +177,14 @@ impl Error {
             _ => None,
         }
     }
+
+    /// Returns the underlying message, if any, that describes this error.
+    pub fn message(&self) -> Option<String> {
+        match &self.inner.kind {
+            ErrorKind::Lex(e) => Some(e.to_string()),
+            ErrorKind::Custom(e) => Some(e.clone()),
```

Comment:
We could return `Cow<str>` here to avoid the `.clone()` if you would prefer.

Reply:
I think lex errors should always have some sort of a static description, could that be exposed? Additionally now that the return value is always `Some` I think the return value here doesn't need `Option`.

Reply:
@alexcrichton Good point about the unnecessary `Option` return. I can remove that.

I don't quite understand what you mean about a static description for `LexError` though. I could factor part of the string description out of the `Display` impl for `LexError`, but since some of the `LexError` cases have arguments (which contribute to the description), I don't see how it can be fully static (since we need to use something like `format!`).

Can you elaborate on what you have in mind for that?

Reply:
Hm I was thinking of returning a static string yeah for all lex errors, but I actually think this makes more sense thinking about it. (you bring up a good point too!)

Seems fine by me in that case!